code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -0,0 +1,543 @@
+
+	ghost.do_jitter_animation(2000) //Animation
+
+	ghost.add_filter("wraith_phase_shift_windup_1", 3, list("type" = "wave", 0, 0, size=rand()*2.5+0.5, offset=rand())) //Cool filter appear
+	ghost.add_filter("wraith_phase_shift_windup_2", 3, list("type" = "wave", 0, 0, size=rand()*2.5+0.5, offset=rand())) //Cool filter appear
+	animate(ghost.get_filter("wraith_phase_shift_windup_1"), x = 60*rand() - 30, y = 60*rand() - 30, size=rand()*2.5+0.5, offset=rand(), time = 0.25 SECONDS, loop = -1, flags=ANIMATION_PARALLEL)
+	animate(ghost.get_filter("wraith_phase_shift_windup_2"), x = 60*rand() - 30, y = 60*rand() - 30, size=rand()*2.5+0.5, offset=rand(), time = 0.25 SECONDS, loop = -1, flags=ANIMATION_PARALLEL)
+
+	if(!do_after(ghost, WRAITH_PHASE_SHIFT_WINDUP, TRUE, ghost, BUSY_ICON_FRIENDLY)) //Channel time/wind up
+		ghost.visible_message("<span class='xenowarning'>[ghost]'s form abruptly consolidates, returning to normalcy.</span>", \
+		"<span class='xenodanger'>We abort our desynchronization.</span>")
+		ghost.remove_filter("wraith_phase_shift_windup_1")
+		ghost.remove_filter("wraith_phase_shift_windup_2")
+		return fail_activate()
+
+	ghost.remove_filter("wraith_phase_shift_windup_1")
+	ghost.remove_filter("wraith_phase_shift_windup_2")
+
+	playsound(owner, "sound/effects/ghost.ogg", 25, 0, 1)
+
+	addtimer(CALLBACK(src, .proc/phase_shift_warning), WRAITH_PHASE_SHIFT_DURATION * WRAITH_PHASE_SHIFT_DURATION_WARNING) //Warn them when Phase Shift is about to end
+	addtimer(CALLBACK(src, .proc/phase_shift_deactivate), WRAITH_PHASE_SHIFT_DURATION)
+	ghost.add_filter("wraith_phase_shift", 4, list("type" = "blur", 5)) //Cool filter appear
+
+	ghost.status_flags = GODMODE | INCORPOREAL //Become temporarily invulnerable and incorporeal
+	ghost.resistance_flags = RESIST_ALL
+	ghost.density = FALSE
+	ghost.throwpass = TRUE
+	ghost.alpha = WRAITH_PHASE_SHIFT_ALPHA //Become translucent
+
+	starting_turf = get_turf(ghost) //Get our starting turf so we can calculate the stun duration later.
+
+	GLOB.round_statistics.wraith_phase_shifts++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "wraith_phase_shifts") //Statistics
+
+	succeed_activate()
+	add_cooldown()
+
+/datum/action/xeno_action/phase_shift/proc/phase_shift_warning()
+
+	if(!owner.get_filter("wraith_phase_shift")) //If phase shift isn't active, cancel out
+		return
+
+	owner.alpha = WRAITH_PHASE_SHIFT_ALPHA * 1.5 //Become less translucent
+
+	to_chat(owner,"<span class='highdanger'>We begin to move back into phase with reality... We can only remain out of phase for [WRAITH_PHASE_SHIFT_DURATION * (1-WRAITH_PHASE_SHIFT_DURATION_WARNING) * 0.1] more seconds!</span>")
+	owner.playsound_local(owner, 'sound/voice/hiss4.ogg', 50, 0, 1)
+
+
+/datum/action/xeno_action/phase_shift/proc/phase_shift_deactivate()
+
+	if(!owner.get_filter("wraith_phase_shift")) //If phase shift isn't active, don't deactivate again.
```

Comment:
will this ever happen

Reply:
It can due to the call from the timer, and the call from resync.